### ISSUE TITLE
LSTM training crashes Python process for long sequences

### ISSUE THREAD

**Issue by sairampillai** [pytorch/pytorch#9621]
## Issue description
Training an LSTM on CPU for long sequences fails with Python process error. 
The sequence length upto 320 is working fine  but anything above that crashes python process with error:
`Process finished with exit code -1073741571 (0xC00000FD)`

The error occurs after the loss function has finished calculating loss.

Debugging python with the Visual studio JIT debugger shows the following unhandled exception:
`Unhandled exception at 0x00007FFCF104E044 (_C.cp36-win_amd64.pyd) in python3.exe: 0xC00000FD: Stack overflow (parameters: 0x0000000000000001, 0x00000003B1203FE8). occurred`

Is this a hardware limitation or can this be solved programatically? 

## Code example
The class used: An RNN followed by a linear layer to output class label distributions.
`class RnnLinearSigmoid(torch.nn.Module):
    def __init__(self, in_dimension, hidden_dimension, out_dimension, rnn_hidden_layers,
                 rnn_activation='relu', batch_size=None, batch_first=False, truncate=5, lstm=False):
        super(RnnLinearSigmoid, self).__init__()
        self.in_dimension = in_dimension
        self.hidden_dimension = hidden_dimension
        self.out_dimension = out_dimension
        self.batch_size = 1 if batch_size is None else batch_size
        self.batch_first = batch_first
        self.hidden_layer = rnn_hidden_layers
        self.is_lstm = lstm
        self.truncate = truncate

        if self.is_lstm:
            self.rnn = torch.nn.LSTM(input_size=in_dimension, hidden_size=hidden_dimension,
                                     num_layers=rnn_hidden_layers, batch_first=batch_first)
        else:
            self.rnn = torch.nn.RNN(input_size=in_dimension, hidden_size=hidden_dimension,
                                    num_layers=rnn_hidden_layers, nonlinearity=rnn_activation, batch_first=batch_first)

        self.linear_layer = torch.nn.Linear(in_features=hidden_dimension, out_features=out_dimension)
        self.activation = torch.nn.Softmax()

    def forward(self, input_tensor):
        if self.is_lstm:
            rnn_output, hidden = self.rnn(input_tensor)
        else:
            hidden = self.init_hidden(self.hidden_layer, self.batch_size, self.hidden_dimension)
            rnn_output, hidden = self.rnn(input_tensor, hidden)

        linear_output = self.linear_layer(rnn_output[-self.truncate:])
        activated_output = linear_output
        return activated_output.view(activated_output.shape[0], self.out_dimension)

    def init_hidden(self, hidden_layers, batch_size, hidden_dimension):
        return torch.zeros(hidden_layers, batch_size, hidden_dimension, requires_grad=True)`

`def train():

    model = RnnLinearSigmoid(in_dimension=100, hidden_dimension=100, out_dimension=5, rnn_hidden_layers=3, lstm=True)

    loss_fn = torch.nn.CrossEntropyLoss()
    optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
    for epoch in range(10000):

        sequence_length = random.randint(320, 4900)
        print(sequence_length)
        input = torch.rand(sequence_length, 1, 100)
        output = model(input)
        answer = np.repeat(np.random.randint(low=0, high=5, size=1), 5)
        answer = torch.from_numpy(answer).long()

        loss = loss_fn(output, answer.detach())
        loss.backward()
        optimizer.step()`

## System Info
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: Could not collect (No)

OS: Microsoft Windows 10 Enterprise
GCC version: (MinGW.org GCC-6.3.0-1) 6.3.0
CMake version: version 3.11.4

Python version: 3.6
Is CUDA available: No
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce 210
Nvidia driver version: 341.95
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy (1.14.5+mkl)
[pip3] torch (0.4.0)
[pip3] torchvision (0.2.1)
[conda] Could not collect (NIL)

**Comment by zou3519** [comment]
The patch is not in 0.4 iirc.

**Comment by SsnL** [comment]
@sairampillai We will release 0.4.1 soon, which likely contains a fix for your problem. When the release happens, please try it and let us know if it solves the bug.

**Comment by sairampillai** [comment]
@SsnL  @zou3519  Thank you for the prompt reply. I will try the update once 0.4.1 is available, and let you know.

**Comment by sairampillai** [comment]
@SsnL @zou3519  The issue has been fixed in PyTorch 0.4.1. Thank you.
